ocaml/database/redo_log.ml

```diff
@@ -886,11 +886,12 @@ let flush_db_exn db log =
     raise (RedoLogFailure "Cannot connect to redo log")
 
 let enable_and_flush db log reason =
-  enable_existing log reason ; flush_db_exn db log
+  enable_existing log reason ;
+  with_lock redo_log_creation_mutex @@ fun () -> flush_db_exn db log
 
 let enable_block_and_flush db log path =
   enable_block_existing log path ;
-  flush_db_exn db log
+  with_lock redo_log_creation_mutex @@ fun () -> flush_db_exn db log
```

Comment:
Why is this mutex held? I see it's only used for manipulating the Set of redo_logs

Reply:
I can test this. This one was suggested by @edwintorok and I tested it.

Reply:
Using `log.mutex` instead causes a deadlock.

Reply:
`with_active_redologs` holds this mutex and is what used to synchronize DB delta writes. Although there are some calls such as shutdown which are not always wrapped with that.